tchannel/tchannel.py

```diff
@@ -176,7 +179,18 @@ def call(self, scheme, service, arg1, arg2=None, arg3=None,
         raise gen.Return(result)
 
     def listen(self, port=None):
-        return self._dep_tchannel.listen(port)
+        with self._listen_lock:
+            if self._dep_tchannel.is_listening():
+                listening_port = int(self._dep_tchannel.hostport.rsplit(":")[1])
```

Comment:
The number of times we're doing this--we should have a `port` attribute exposed. Let's do that in a separate PR later.